openslides/core/static/js/core/site.js

```diff
@@ -661,6 +661,83 @@ angular.module('OpenSlidesApp.core.site', [
     }
 ])
 
+/* change:
+ *  Callback if file changes. The one parameter is csv passing the parsed file
+ * config:
+ * - accept: String with extensions: default '.csv .txt'
+ * - encodingOptions: List with encodings. Default ['UTF-8', 'ISO-8859-1']
+ * - parseConfig: a dict passed to PapaParse
+ */
+.directive('csvImport', [
+    function () {
+        return {
+            restrict: 'E',
+            templateUrl: 'static/templates/csv-import.html',
+            scope: {
+                change: '&',
+                config: '=?',
+            },
+            controller: function ($scope, $element, $attrs, $location) {
+                // set config if it is not given
+                if (!$scope.config) {
+                    $scope.config = {};
+                }
+                if (!$scope.config.parseConfig) {
+                    $scope.config.parseConfig = {};
+                }
+
+                $scope.inputElement = angular.element(document.querySelector('#csvFileSelector'));
```

Comment:
This looks like that we query for an id in the whole document. So it is not possible to use this directive more than once in a template, is it? I propose to try to find the file element by looking inside the `$element` and using a specific class instead of an id.